templates/spellcheck_test.tt.html

```diff
@@ -0,0 +1,37 @@
+<p>The spellcheck results are based on matches in a given taxonomy.</p>
+<p>For words with at least 5 characters, we consider an edit distance of 1 (insert, replace or delete).</p>
+
+[% IF action_process == 'process' %]
+    <p>
+        Spellcheck result for taxonomy <strong>[% tagtype %]</strong> and language <strong>[% lc %]</strong>:
+    </p>
+    <table>
+        <caption>Spellcheck Results</caption>
+        <tr>
+            <th scope="col">Input</th>
+            <th scope="col">Correction</th>
+            <th scope="col">[% lc %] tag</th>
+            <th scope="col">Canonical tag</th>
+        </tr>
+        [% FOREACH token IN tokens %]
+            <tr>
+                <td>[% token.token %]</td>
+                <td>[% token.tag %]</td>
+                <td>[% token.tagid %]</td>
+                <td>[% token.canon_tagid %]</td>
+            </tr>
+        [% END %]
+    </table>
+[% END %]
+
+[% IF action_display == 'display' %]
+    <form method="get" action="/cgi/spellcheck_test.pl" enctype="multipart/form-data">
+        Taxonomy: <input type="text" name="tagtype" id="tagtype" value="[% tagtype %]" /><br /><br />
+        Text (language code: [% lc %]): <br/>
+
+        <textarea id="text" name="text" style="height:8rem;">[% text %]</textarea>
+        <input type="hidden" name="type" value="add"  />
+        <input type="hidden" name="action" value="process"  />
+        <input type="submit" name=".submit" />
```

Comment:
You could add `class="button"` to apply the same button style as is applied to other buttons on the site. I'm not sure how important this would be, considering it's a test page for debugging purposed. 🙂

Reply:
```suggestion
        <input class="button" type="submit" name=".submit" />
```